mmdet/datasets/openimages.py

```diff
@@ -0,0 +1,829 @@
+        # TODO: set flag without width and height
+
+    def get_father(self, hierarchy_file):
+        """Get hierarchy for classes.
+
+        Args:
+            hierarchy_file (sty): File path to the hierarchy for classes.
+
+        Returns:
+            ndarray: The matrix of the corresponding relationship between
+            the father class and the child class, of shape
+            (class_num, class_num).
+        """
+
+        assert hierarchy_file.endswith('json')
+        if self.data_root is not None:
+            if not osp.isabs(hierarchy_file):
+                hierarchy_file = osp.join(self.data_root, hierarchy_file)
+        with open(hierarchy_file, 'r') as f:
+            hierarchy = json.load(f)
+        class_num = len(self.CLASSES)
+        class_label_tree = np.eye(class_num, class_num)
+        class_label_tree = self.get_parent_children(hierarchy,
+                                                    class_label_tree)
+        return class_label_tree
+
+    def get_parent_children(self,
+                            hierarchy,
+                            class_label_tree,
+                            father=[],
+                            get_all_fathers=True):
+        """Get matrix of the corresponding relationship between the father
+        class and the child class.
+
+        Args:
+            hierarchy (dict): Including label name and corresponding
+                subcategory. Keys of dicts are:
+
+                - `LabeName` (str): Name of the label.
+                - `Subcategory` (dict | list): Corresponding subcategory(ies).
+            class_label_tree (ndarray): The matrix of the corresponding
+                relationship between the father class and the child class,
+                of shape (class_num, class_num).
+            father (list): Corresponding father class.
+            get_all_fathers (bool): Whether get all father name. Default: True
+
+        Returns:
+            ndarray: The matrix of the corresponding relationship between
+            the father class and the child class, of shape
+            (class_num, class_num).
+        """
+
+        if 'Subcategory' in hierarchy:
+            for node in hierarchy['Subcategory']:
+                if 'LabelName' in node:
+                    children_name = node['LabelName']
+                    children_index = self.index_dict[children_name]
+                    children = [children_index]
+                else:
+                    continue
+                if len(father) > 0:
+                    for father_index in father:
+                        if get_all_fathers:
+                            children.append(father_index)
+                        class_label_tree[children_index, father_index] = 1
+
+                class_label_tree = self.get_parent_children(
+                    node, class_label_tree, father=children)
+
+        return class_label_tree
+
+    def get_gt_fathers(self, annotations):
```

Comment:
rename this function

Reply:
How about get_gt_parents?

Reply:
the return value is `annotations`, how about `get_parents_ann`?

Reply:
Done